pkg/kp/kv.go

```diff
@@ -0,0 +1,132 @@
+// Package kp provides a generalized API for reading and writing pod manifests
+// in consul.
+package kp
+
+import (
+	"bytes"
+	"time"
+
+	"github.com/armon/consul-api"
+	"github.com/square/p2/pkg/pods"
+	"github.com/square/p2/pkg/util"
+)
+
+type ManifestResult struct {
+	Manifest pods.PodManifest
+	Path     string
+}
+type Options struct {
+	Token   string
+	Address string
+}
+
+type Store struct {
+	client *consulapi.Client
+}
+
+func NewStore(opts Options) *Store {
+	conf := consulapi.DefaultConfig()
+	conf.Address = opts.Address
+	conf.Token = opts.Token
+
+	// the error is always nil
```

Comment:
The package we rely on is deprecated (in favor of `github.com/hashicorp/consul/api`) so it's unlikely to change. The newer version of the package also returns a nil error every time (at least right now).

I don't like ignoring errors, but that makes superfluous errors all the more wasteful. Personally I find it really dumb to push the error down to every client of this package when it's always going to be nil... if we move to the new package we can consider this more carefully, but I don't think there should be an error return on consulapi.NewClient in the first place.